longevity_tombstone_gc_test.py

```diff
@@ -0,0 +1,125 @@
+from sdcm.utils.common import ParallelObject
+from sdcm.utils.sstable_utils import SstableUtils
+
+
+class TombstoneGcLongevityTest(TWCSLongevityTest):
+    keyspace = 'scylla_bench'
+    table = 'test'
+    ks_cf = f'{keyspace}.{table}'
+    propagation_delay = 60 * 4  # setting a value shorter than the default, of 4 minutes
+    repair_date = None
+    db_node = None
+
+    def _run_repair_and_major_compaction(self, wait_propagation_delay: bool = False):
+        self.log.info('Run a flush for user-table on nodes')
+        triggers = [partial(node.run_nodetool, sub_cmd=f"flush -- {self.keyspace}", ) for node in self.db_cluster.nodes]
+        ParallelObject(objects=triggers, timeout=1200).call_objects()
+        # for node in self.db_cluster.nodes:
+        #     node.run_nodetool(f"flush -- {self.keyspace}")
+
+        self.log.info('Run a repair for user-table on nodes')
+        triggers = [partial(node.run_nodetool, sub_cmd="repair", args=f"-pr {self.keyspace} {self.table}", ) for node
+                    in self.db_cluster.nodes]
+        ParallelObject(objects=triggers, timeout=1200).call_objects()
+        # for node in self.db_cluster.nodes:
+        #     node.run_nodetool(sub_cmd="repair", args=f"-pr {self.keyspace} {self.table}")
+
+        if wait_propagation_delay:
+            time.sleep(self.propagation_delay)
+        self.repair_date = datetime.datetime.now()
+        self.log.info('Run a major compaction for user-table on node')
+        self.db_node.run_nodetool("compact", args=f"{self.keyspace} {self.table}")
+        self.wait_no_compactions_running()
+
+    def test_switch_tombstone_gc_modes(self):
+        """
+        test steps:
+        -----------
+        Based on TWCS TTL 48h longevity configuration with a much shorter TTL (few minutes) and same for gc-grace-seconds.
+        Start with tombstone-gc-mode disabled
+        wait a duration of TTL.
+        count number of tombstones.
+        run a repair + a major compaction.
+        count again to see not fewer tombstones exit.
+        change gc-grace-seconds back to default of 10 days.
+        change gc-mode to repair
+        run a repair + wait propagation delay + run a major compaction.
+        verify no tombstones exist in post-repair-created sstables.
+        change gc-mode to immediate.
+        wait for load to end + TTL period.
+        Run a major compaction.
+        Verify no tombstones.
+        """
+        # pylint: disable=too-many-locals
+
+        self.create_tables_for_scylla_bench()
+        self.db_node = self.db_cluster.nodes[0]
+        self.run_post_prepare_cql_cmds()
+        stress_queue = []
+
+        stress_cmd = self.params.get('stress_cmd')
+        params = {'stress_cmd': stress_cmd, 'round_robin': self.params.get('round_robin')}
+        self._run_all_stress_cmds(stress_queue, params)
+
+        self.log.info('Wait a duration of TTL + propagation_delay_in_seconds')
+        wait_for_tombstones = 4 * 60 * 2
```

Comment:
where is the TTL set?
also, the `propagation_delay_in_seconds` is a param, so it should use it, instead of hard coded `60 * 4` on variable `self.propagation_delay`

Reply:
not addressed

Reply:
fixed